modules/rpc/yarpc.go

```diff
@@ -105,78 +176,105 @@ func newYarpcModule(
 		}
 	}
 
-	err := module.Host().Config().Get(fmt.Sprintf("modules.%s", module.Name())).PopulateStruct(cfg)
-	// found values, update module
-	module.config = *cfg
+	err = module.Host().Config().Scope("modules").Get(module.Name()).PopulateStruct(&module.config)
+	if err != nil {
+		return module, err
+	}
+
+	module.config.inboundMiddleware = inboundMiddlewareFromCreateInfo(mi)
+	module.config.onewayInboundMiddleware = onewayInboundMiddlewareFromCreateInfo(mi)
+
+	tchTransport, err := tch.NewChannelTransport(
+		tch.ServiceName(module.config.AdvertiseName),
+		tch.ListenAddr(module.config.Bind),
+	)
 
-	module.inboundMiddlewares = inboundMiddlewaresFromCreateInfo(mi)
-	module.onewayInboundMiddlewares = onewayInboundMiddlewaresFromCreateInfo(mi)
+	module.config.inbounds = []transport.Inbound{tchTransport.NewInbound()}
 
 	return module, err
 }
 
-// Start begins serving requests over YARPC
-func (m *YarpcModule) Start(readyCh chan<- struct{}) <-chan error {
+// Start begins serving requests over RPC. It first stops the current dispatcher, merges configs from the global
+// collection and then starts the new dispatcher.
+func (m *YARPCModule) Start(readyCh chan<- struct{}) <-chan error {
+	if m.IsRunning() {
+		panic("module is already running")
+	}
+
 	m.stateMu.Lock()
 	defer m.stateMu.Unlock()
 
-	interceptor := yarpc.UnaryInboundMiddleware(m.inboundMiddlewares...)
-	onewayInterceptor := yarpc.OnewayInboundMiddleware(m.onewayInboundMiddlewares...)
-
 	ret := make(chan error, 1)
-	tchTransport, err := tch.NewChannelTransport(
-		tch.ServiceName(m.config.AdvertiseName),
-		tch.ListenAddr(m.config.Bind),
-	)
-	if err != nil {
-		ret <- errors.New("Unable to create TChannel transport " + err.Error())
+	if err := _configs.addConfig(&m.config); err != nil {
+		ret <- errors.New("can't add module config " + err.Error())
 		return ret
 	}
 
-	m.rpc, err = _dispatcherFn(m.Host(), yarpc.Config{
-		Name: m.config.AdvertiseName,
-		Inbounds: []transport.Inbound{
-			tchTransport.NewInbound(),
-		},
-		InboundMiddleware: yarpc.InboundMiddleware{
-			Unary:  interceptor,
-			Oneway: onewayInterceptor,
-		},
-		Tracer: m.Tracer(),
-	})
+	config, err := _configs.mergeConfigs()
+
+	if err != nil {
+		ret <- errors.New("unable to merge configs from multiple modules " + err.Error())
```

Comment:
maybe a colon after `modules` to show that the rest is the error.

also, should this be `errors.Wrap` instead?